### PR TITLE
feat(components): Add file preview to dropzone component

### PR DESCRIPTION
**Pull request by markflorkowski**

### CODE REVIEW

File: packages/react/src/components/dropzone.tsx [hunk 1]
```diff
@@ -200,27 +282,39 @@ export function UploadDropzone<TRouter extends FileRouter>(
...
+          ) : (
+            `Loading...`
+          ))}
         <input className="sr-only" {...getInputProps()} />
       </label>
-      <div
-        className={twMerge(
-          "m-0 h-[1.25rem] text-xs leading-5 text-gray-600",
-          styleFieldToClassName(
+      {$props.config?.preview !== false && files.length > 0 ? null : (
```
Comment (Mr0Bread) [inline packages/react/src/components/dropzone.tsx:296]: Why hide allowed content if file preview is enabled?

**Comment by Mr0Bread** [comment]
Are you planning on adding custom styling of File Preview cards in this PR or in separate one?

**Comment by markflorkowski** [comment]
> Are you planning on adding custom styling of File Preview cards in this PR or in separate one?

Will do it in this one, I just need to finalize what I want the default previews to look like first before figuring out how to expose styling

https://twitter.com/r_marked/status/1700244881854362013 😅

**Comment by Mr0Bread** [comment]
> > Are you planning on adding custom styling of File Preview cards in this PR or in separate one?
> 
> Will do it in this one, I just need to finalize what I want the default previews to look like first before figuring out how to expose styling
> 
> https://twitter.com/r_marked/status/1700244881854362013 😅

Posted several questions/ideas

File: packages/react/src/components/dropzone.tsx [hunk 2]
```diff
@@ -46,18 +46,90 @@ export type UploadDropzoneProps<TRouter extends FileRouter> =
...
+}: {
+  files: File[];
+  updateFiles: React.Dispatch<React.SetStateAction<File[]>>;
+}) => {
+  return (
+    <div className="flex flex-row flex-wrap items-center justify-center gap-2">
+      {files.map((file) => (
+        <div
+          key={file.name}
+          className="group relative flex h-24 w-24 flex-col items-center justify-center rounded-lg  p-1 ring-1 ring-gray-200 sm:h-32 sm:w-32"
```
Comment (zmzlois) [inline packages/react/src/components/dropzone.tsx:66]: maybe `h-full w-auto rounded-sm` 
<IconX className="absolute top-5 left-5">
enlarge the preview and take up more space will look better?

File: packages/react/src/components/dropzone.tsx [hunk 3]
```diff
@@ -46,18 +46,90 @@ export type UploadDropzoneProps<TRouter extends FileRouter> =
...
 
+const FilesPreview = ({
+  files,
+  updateFiles,
+}: {
+  files: File[];
+  updateFiles: React.Dispatch<React.SetStateAction<File[]>>;
+}) => {
+  return (
+    <div className="flex flex-row flex-wrap items-center justify-center gap-2">
```
Comment (zmzlois) [inline packages/react/src/components/dropzone.tsx:62]: bg-zinc-200?

File: packages/react/src/components/dropzone.tsx [hunk 4]
```diff
@@ -46,18 +46,90 @@ export type UploadDropzoneProps<TRouter extends FileRouter> =
...
+          className="group relative flex h-24 w-24 flex-col items-center justify-center rounded-lg  p-1 ring-1 ring-gray-200 sm:h-32 sm:w-32"
+          onClick={(e) => {
+            e.stopPropagation();
+            updateFiles(
+              files.filter((f) => f.name !== file.name && f.size !== file.size),
+            );
+          }}
+        >
+          {/* Hover delete overlay */}
+          <div className="absolute left-0 top-0 flex h-full w-full items-center justify-center rounded-lg bg-red-600/60 text-white opacity-0 group-hover:opacity-100">
```
Comment (zmzlois) [inline packages/react/src/components/dropzone.tsx:75]: this ... doesn't look nice

File: packages/react/src/components/dropzone.tsx (hunk 2 above)
Comment (zmzlois) [inline packages/react/src/components/dropzone.tsx:66]: maybe `h-full w-auto rounded-sm` 
`<IconX className="absolute top-5 left-5 group-hover:text-red-600 text-red-400">`
enlarge the preview and take up more space will look better?

**Review by zmzlois — COMMENTED** [review]
if the whole preview button looks bigger it will look less...industrial

File: packages/react/src/components/dropzone.tsx [hunk 5]
```diff
@@ -72,11 +72,11 @@ const FilesPreview = ({
           }}
         >
           {/* Hover delete overlay */}
-          <div className="absolute left-0 top-0 flex h-full w-full items-center justify-center rounded-lg bg-red-600/60 text-white opacity-0 group-hover:opacity-100">
+          <div className="absolute left-0 top-0 flex h-full w-full items-center justify-center rounded-lg bg-black/60 text-white opacity-0 group-hover:opacity-100">
```
Comment (zmzlois) [inline packages/react/src/components/dropzone.tsx:75]: Much better, red makes me feel like I am having a heart attack and black doesn't

File: packages/react/src/components/dropzone.tsx [hunk 6]
```diff
@@ -59,11 +59,11 @@ const FilesPreview = ({
   updateFiles: React.Dispatch<React.SetStateAction<File[]>>;
 }) => {
   return (
-    <div className="flex flex-row flex-wrap items-center justify-center gap-2">
+    <div className="flex h-36 w-full flex-row gap-2 overflow-x-auto">
```
Comment (zmzlois) [inline packages/react/src/components/dropzone.tsx:62]: `hover:cursor-pointer hover:ring-1` or `hover:bg-zinc-1` ?

**Review by zmzlois — COMMENTED** [review]
beautiful

**Comment by markflorkowski** [comment]
Closing for now as I haven't been able to come up with something that I feel is universal enough while remaining minimal.

Leaning towards not including previews in the component, and providing examples of custom components with this functionality that are more tailored to specific applications (eg.  photo gallery, document lists, etc.)

**Pull request closed without merging**